x-pack/plugins/log_explorer/public/components/flyout_detail/sub_components/log_level.tsx

```diff
@@ -6,27 +6,55 @@
  */
 
 import React from 'react';
-import { EuiBadge, type EuiBadgeProps } from '@elastic/eui';
+import { useEuiTheme } from '@elastic/eui';
 import { FlyoutDoc } from '../types';
+import { ChipWithPopover } from '../../common/popover_chip';
+import * as constants from '../../../../common/constants';
 
-const LEVEL_DICT: Record<string, EuiBadgeProps['color']> = {
+const LEVEL_DICT = {
   error: 'danger',
   warn: 'warning',
   info: 'primary',
-  default: 'default',
-};
+  debug: 'accent',
+} as const;
 
 interface LogLevelProps {
   level: FlyoutDoc['log.level'];
+  dataTestSubj?: string;
+  renderInFlyout?: boolean;
 }
 
-export function LogLevel({ level }: LogLevelProps) {
+export function LogLevel({
+  level,
+  dataTestSubj = 'logExplorerFlyoutLogLevel',
+  renderInFlyout = false,
+}: LogLevelProps) {
+  const { euiTheme } = useEuiTheme();
   if (!level) return null;
-  const levelColor = LEVEL_DICT[level] ?? LEVEL_DICT.default;
+  const levelColor = LEVEL_DICT[level as keyof typeof LEVEL_DICT]
+    ? euiTheme.colors[LEVEL_DICT[level as keyof typeof LEVEL_DICT]]
+    : null;
+
+  if (renderInFlyout) {
```

Comment:
Given the two branches are almost equal in size, would it make sense to create two individual components instead of one with a boolean flag?

Reply:
I initially built 2 separate components, `log_level_flyout` and `log_level_table`. I found both them sharing almost the same code with only 2 difference.

1. Flyout Log Level will not have icons
2. Flyout Log level should not be a clickable item as it has its own popover on hover. 

Given these 2 challenges can be prop controlled i chose to branch them out. But lmk, if you still think we should split them